### PR TITLE
new sensor full image calculations

### PR DESCRIPTION
**Pull request by tylerjw**
now working reading full images with timing debugging

### CODE REVIEW

File: firmware/mlx90641/MLX90641_API.c [hunk 1]
```diff
@@ -179,32 +199,100 @@ int HammingDecode(uint16_t *eeData)
...
+                        return -10;
         }
-        subPage = statusRegister & 0x0001;
+        subPage = status_register & 0x0001;
+
+        ready = chVTGetSystemTime();
 
         while(dataReady != 0 && cnt < 5) {
-                error = MLX90641_I2CWrite(slaveAddr, 0x8000, 0x0030);
+                error = MLX90641_I2CWrite(slaveAddr, 0x8000, (status_register & 0xFFF7));
```
Comment (brentpicasso) [inline firmware/mlx90641/MLX90641_API.c]: 0x0030 is what the OEM library has originally, but this follows the datasheet?

File: firmware/mlx90641/MLX90641_API.c [hunk 2]
```diff
@@ -179,32 +199,100 @@ int HammingDecode(uint16_t *eeData)
...
                 if(error == -1) {
                         return error;
                 }
 
                 if(subPage == 0) {
+                        // log_info(_LOG_PFX "GetFrameData Subpage 0\r\n");
```
Comment (brentpicasso) [inline firmware/mlx90641/MLX90641_API.c]: remove commented out code

File: firmware/mlx90641/MLX90641_API.c [hunk 3]
```diff
@@ -230,6 +318,7 @@ int MLX90641_GetFrameData(uint8_t slaveAddr, uint16_t *frameData)
                                 return error;
                         }
                 } else {
+                        // log_info(_LOG_PFX "GetFrameData Subpage 1\r\n");
```
Comment (brentpicasso) [inline firmware/mlx90641/MLX90641_API.c]: remove commented out code

**Comment by brentpicasso** [comment]
looks good. only minor things; ensure warnings are fixed and we can merge

File: firmware/mlx90641/MLX90641_API.c (hunk 1 above)
Comment (tylerjw) [reply firmware/mlx90641/MLX90641_API.c]: yes

**Comment by tylerjw** [comment]
```
main.c:73:13: warning: '_start_watchdog' defined but not used [-Wunused-function]
 static void _start_watchdog(void)
```
that's the only warning

**Pull request merged**